server/app/services/apikey/ApiKeyService.java

```diff
@@ -206,7 +205,10 @@ private ApiKeyGrants resolveGrants(DynamicForm form) throws ProgramNotFoundExcep
       String programSlug = matcher.group(1);
 
       if (!programSlugs.contains(programSlug)) {
-        throw new ProgramNotFoundException(programSlug);
+        // Making the root exception cause a ProgramNotFoundException will cause
+        // controllers.ErrorHandler to handle it
+        // as a client-error, rather than server-error.
+        throw new RuntimeException(new ProgramNotFoundException(programSlug));
```

Comment:
discussion: This is fine, though subsequently, I think it'd be good/fine to make ProgramNotFoundException a Runtime Exception.  If code really needs to know it can do a hasProgramId() kinda check, but I'm guessing most just catch and wrap this as notFound without doing any other recovery or reporting.

Reply:
Yeah, to make the change we'll need to take a look at all the places `ProgramNotFoundException` is being used and see if any of them are using it for control flow that's fundamentally different from what the default behavior with the new pattern does. If so we should probably leave it as a checked exception.